### PR TITLE
Allow overlapping activities

### PR DESCRIPTION
**Pull request by jmthomas**

### CODE REVIEW

**Review by ryanmelt — CHANGES_REQUESTED** [review]
I think the time validation should be optional with an argument provided to create/update .  It can default to don't validate, but sometimes I think it would a nice option to guarantee no overlaps.

**Comment by ryanmelt** [comment]
Overlap option looks great.   The other thing I'm worried about is ordering if we have multiple items in the same second.  Is there anyway we can enforce ordering?

**Comment by jmthomas** [comment]
Should we just allow sub-second start time and force the user to ensure ordering?

**Comment by ryanmelt** [comment]
I think that would work.  If two items have the exact same time, that implies that the order shouldn't matter.

File: openc3/lib/openc3/models/activity_model.rb [hunk 1]
```diff
@@ -168,7 +168,8 @@ def initialize(
     #
     # @param [Integer] ignore_score - should be nil unless you want to ignore a time when doing an update
     def validate_time(ignore_score = nil)
-      array = Store.zrevrangebyscore(@primary_key, @stop - 1, @start - MAX_DURATION)
+      # Adding a '(' makes the max value exclusive
+      array = Store.zrevrangebyscore(@primary_key, "(#{@stop}", @start - MAX_DURATION)
```
Comment (ryanmelt) [inline openc3/lib/openc3/models/activity_model.rb:172]: What is going on with the weird parentheses in the quoted string here?

Comment (ryanmelt) [reply openc3/lib/openc3/models/activity_model.rb:172]: Nevermind - Nice comment!

**Review by ryanmelt — APPROVED** [review]

**Pull request merged**